tmk_core/common/report.h

```diff
@@ -159,10 +159,10 @@ typedef struct {
 
 typedef struct {
     uint8_t buttons;
-    int8_t x;
-    int8_t y;
-    int8_t v;
-    int8_t h;
+    int16_t x;
+    int16_t y;
+    int16_t v;
+    int16_t h;
```

Comment:
I would assume updating this has implications across the different protocols because each descriptor would need to be updated for the larger integer size. I just modified this in place just to test with adb_usb which uses the lufa protocol. I'm not sure on the best approach you want to take here. I was thinking either a new `report_mouse_extended_t` datatype for the 16 bit values and conditionally enable it, or increase all of the descriptors and usages of `report_mouse_t` to 16 bit.

Reply:
I dont' like to change codes under tmk_core if possible.

What's is actually the point of the extended report? This is requirement for this scrolling feature?

Reply:
Ah, it seems that Turbo Mouse replies with 10-bit data for XY movement.
The extended mouse report to 16-bit makes sense for X and Y at least now.

But I will have to confirm how mouse report extension effects other projects before merge this PR.

Thanks for your contribution!